src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -222,11 +222,12 @@ public async ValueTask SetupAsync(CancellationToken cancellationToken)
                 BinaryPrimitives.WriteUInt32BigEndian(_outgoingBuffer.AvailableSpan, windowUpdateAmount);
                 _outgoingBuffer.Commit(4);
 
+                _receivedSettingsAck = false;
+                _ = ProcessIncomingFramesAsync();
```

Comment:
Can you please add a comment here explaining that calling `ProcessIncoming` before `WriteAsync` is an intentional implementation detail to support named pipes?

Reply:
How does this sound?

```csharp
// Processing the incoming frames before sending the client preface and SETTINGS is necessary when using a NamedPipe as a transport.
// If the preface and SETTINGS coming from the server are not read first the below WriteAsync and the ProcessIncomingFramesAsync fall into a deadlock.
```

Also, there are 2 tests that I could clone and add a test with a NamedPipe stream.

ConnectCallback_ConnectionPrefix_Success clone and add **ConnectCallback_NamedPipeConnectionPrefix_Success**
ConnectCallback_UseUnixDomainSocket_Success clone and add **ConnectCallback_UseNamedPipeSocket_Success**

would that help? If the order of things ever change those tests could easily catch the issue.

Reply:
That looks good to me.
The tests would be appreciated to prevent future regressions.

Reply:
Hi @MihaZupan  I have created a test that does detect the regresion, the only problem is that it does fail for HTTP1 since I didn't fix anything regarding HTTP1. So I guess the HTTP1 connection would need a similar fix. 
Should I work on that fix too or can the test target only http2 via a test attribute on the method?


<img width="845" alt="image" src="https://user-images.githubusercontent.com/30415120/182974960-58537eea-1fb8-4f4a-8df4-ead89e177851.png">

Reply:
IMO no one has asked for http/1.1 so leave it unchanged. Support can be added in the future when a need is identified.

Reply:
I modified the test so it would only be part of the http2 test flow only.
If a regression gets introduced the ConnectCallback_UseNamedPipe_Success test would fail.
I have tested that by reverting my change and the test does detect the regression.

Reply:
Just as an FYI I tested the new library on my test project and it works like a charm too :) Thanks a lot @MihaZupan  and @JamesNK  for your help